clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -779,8 +779,7 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
     @SafeVarargs
     public static <T> Set<T> mkSet(T... elems) {
         Set<T> result = new HashSet<>((int) (elems.length / 0.75) + 1);
-        for (T elem : elems)
-            result.add(elem);
+        result.addAll(Arrays.asList(elems));
```

Comment:
we are creating an intermediate data structure (list) here in the new code. It might not be worth it.

(same for other places where we use Arrays.asList)

Reply:
Yeah, the tests failed yesterday, so I will be reverting that particular commit